scripts/testnet/sovereignBridge/config/esdt-safe.snippets.sh

```diff
@@ -2,167 +2,225 @@ ESDT_SAFE_ADDRESS=$(mxpy data load --partition=${CHAIN_ID} --key=address-esdt-sa
-    mxpy --verbose contract deploy \
+    echo "Deploying ESDT Safe contract on main chain..."
+
+    local OUTFILE="${OUTFILE_PATH}/deploy-esdt-safe.interaction.json"
+    mxpy contract deploy \
         --bytecode=$(eval echo ${ESDT_SAFE_WASM}) \
         --pem=${WALLET} \
         --proxy=${PROXY} \
         --chain=${CHAIN_ID} \
         --gas-limit=200000000 \
         --arguments \
             false \
-            ${MIN_VALID_SIGNERS} \
-            ${INITIATOR_ADDRESS} \
-            ${SIGNERS} \
-        --outfile="${SCRIPT_PATH}/deploy-esdt-safe.interaction.json" \
+        --outfile=${OUTFILE} \
         --recall-nonce \
         --wait-result \
         --send || return
 
-    local TX_STATUS=$(mxpy data parse --file="${SCRIPT_PATH}/deploy-esdt-safe.interaction.json"  --expression="data['transactionOnNetwork']['status']")
-    if [ "$TX_STATUS" != "success" ]; then
-        echo "Transaction was not successful"
-        return
-    fi
+    printTxStatus ${OUTFILE} || return
 
-    local ADDRESS=$(mxpy data parse --file="${SCRIPT_PATH}/deploy-esdt-safe.interaction.json"  --expression="data['contractAddress']")
+    local ADDRESS=$(mxpy data parse --file=${OUTFILE} --expression="data['contractAddress']")
     mxpy data store --partition=${CHAIN_ID} --key=address-esdt-safe-contract --value=${ADDRESS}
     ESDT_SAFE_ADDRESS=$(mxpy data load --partition=${CHAIN_ID} --key=address-esdt-safe-contract)
     echo -e "ESDT Safe contract: ${ADDRESS}"
 
     local SOVEREIGN_CONTRACT_ADDRESS=$(computeFirstSovereignContractAddress)
     mxpy data store --partition=sovereign --key=address-esdt-safe-contract --value=${SOVEREIGN_CONTRACT_ADDRESS}
     ESDT_SAFE_ADDRESS_SOVEREIGN=$(mxpy data load --partition=sovereign --key=address-esdt-safe-contract)
-    echo -e "ESDT Safe sovereign contract: ${SOVEREIGN_CONTRACT_ADDRESS}"
+    echo -e "ESDT Safe sovereign contract: ${SOVEREIGN_CONTRACT_ADDRESS}\n"
 }
 
 upgradeEsdtSafeContract() {
-    mxpy --verbose contract upgrade ${ESDT_SAFE_ADDRESS} \
+    echo "Upgrading ESDT Safe contract on main chain..."
+    checkVariables ESDT_SAFE_ADDRESS || return
+
+    local OUTFILE="${OUTFILE_PATH}/upgrade-esdt-safe.interaction.json"
+    upgradeEsdtSafeContractCall ${ESDT_SAFE_ADDRESS} ${PROXY} ${CHAIN_ID} ${OUTFILE}
+}
+
+upgradeEsdtSafeContractSovereign() {
+    echo "Upgrading ESDT Safe contract on sovereign chain..."
+    checkVariables ESDT_SAFE_ADDRESS_SOVEREIGN || return
+
+    local OUTFILE="${OUTFILE_PATH}/upgrade-esdt-safe-sovereign.interaction.json"
+    upgradeEsdtSafeContractCall ${ESDT_SAFE_ADDRESS_SOVEREIGN} ${PROXY_SOVEREIGN} ${CHAIN_ID_SOVEREIGN} ${OUTFILE}
+}
+
+upgradeEsdtSafeContractCall() {
+    if [ $# -lt 4 ]; then
+        echo "Usage: $0 <arg1> <arg2> <arg3> <arg4>"
+        exit 1
+    fi
+
+    local ADDRESS=$1
+    local URL=$2
+    local CHAIN=$3
+    local OUTFILE=$4
+
+    mxpy contract upgrade ${ADDRESS} \
         --bytecode=$(eval echo ${ESDT_SAFE_WASM}) \
```

Comment:
Is this wasm file same for both main/sovereign chain?

Reply:
Currently, yes. Same wasm is for both chains.